### ISSUE TITLE
Transition to numpy default_rng as provider for random functions

### ISSUE THREAD

**Issue by mlincett** [icecube/flarestack#129]
According to current [numpy documentation](https://numpy.org/doc/stable/reference/random/index.html), random functions should now use an instance of `default_rng` rather than `numpy.random` direct methods.

**Comment by robertdstein** [comment]
Seems reasonable!

**Comment by mlincett** [comment]
> Seems reasonable!

Yes, we already agreed on that! I have just been too lazy/busy to fix this in the meantime, but let's keep it on my watch.
